### ISSUE TITLE
Verify Eclipse IDE experience with projects using bti infrastructure

### ISSUE THREAD

**Issue by aclement** [spring-projects-experimental/spring-native#473]

**Comment by aclement** [comment]
If i import a sample into the IDE, I get a big popup box with an error:
![image](https://user-images.githubusercontent.com/226298/108281636-f950eb00-7134-11eb-9794-a0013076af0e.png)

**Comment by aclement** [comment]
This section added to the pom.xml will cause the IDE to ignore the issue:
```
<build>
 ...
<pluginManagement>
    <plugins>
        <plugin>
            <groupId>org.eclipse.m2e</groupId>
            <artifactId>lifecycle-mapping</artifactId>
            <version>1.0.0</version>
            <configuration>
                <lifecycleMappingMetadata>
                    <pluginExecutions>
                        <pluginExecution>
                            <pluginExecutionFilter>
                                 <groupId>org.springframework.experimental</groupId>
                                <artifactId>spring-native-maven-plugin</artifactId>
                                <versionRange>[0.0.0,)</versionRange>
                                <goals>
                                    <goal>bootstrapTests</goal>
                                </goals>
                            </pluginExecutionFilter>
                            <action>
                                <ignore />
                            </action>
                        </pluginExecution>
                    </pluginExecutions>
                </lifecycleMappingMetadata>
            </configuration>
        </plugin>
    </plugins>
</pluginManagement>
</build>
```

**Comment by aclement** [comment]
With that added I can build the project, it doesn't have errors on it. It does generate the spring factories code. I don't have a great debugging experience due to #474 - can the plugin configuration above go into the metadata of our maven plugin?

**Comment by aclement** [comment]
(although I do worry ignoring bootstrapTests like that above is then not generating the springfactories for test purposes)

**Comment by sdeleuze** [comment]
FYI we made some fixes on running test recently on fixing how sources and resources are configured in tests (in addition to the plugin rename). Looking forward progresses on #474 as well to move forward on the IDE experience.

**Comment by sdeleuze** [comment]
Please check again with [that change](https://github.com/spring-projects-experimental/spring-native/commit/edc26826872550a5add6f3a2f370a09c8093cd87#diff-356f1a83803292772a6a2875e3f9fb1aa2ecb8e6dc8b80e3994df9e75e4dadceR8) if that now works in Eclipse out of the box.

**Comment by aclement** [comment]
The changes do prevent the popup now. But on import I have hundreds (216) compile errors (webmvc-tomcat). 
```
Description	Resource	Path	Location	Type
BeanCurrentlyInCreationFailureAnalyzer cannot be resolved to a type	_FactoryProvider.java	/webmvc-tomcat/target/spring-aot-test/src/main/java/org/springframework/boot/diagnostics/analyzer	line 4	Java Problem
BeanCurrentlyInCreationFailureAnalyzer cannot be resolved to a type	_FactoryProvider.java	/webmvc-tomcat/target/spring-aot-test/src/main/java/org/springframework/boot/diagnostics/analyzer	line 5	Java Problem
BeanCurrentlyInCreationFailureAnalyzer cannot be resolved to a type	_FactoryProvider.java	/webmvc-tomcat/target/spring-aot/src/main/java/org/springframework/boot/diagnostics/analyzer	line 4	Java Problem
BeanCurrentlyInCreationFailureAnalyzer cannot be resolved to a type	_FactoryProvider.java	/webmvc-tomcat/target/spring-aot/src/main/java/org/springframework/boot/diagnostics/analyzer	line 5	Java Problem
BeanDefinitionOverrideFailureAnalyzer cannot be resolved to a type	_FactoryProvider.java	/webmvc-tomcat/target/spring-aot-test/src/main/java/org/springframework/boot/diagnostics/analyzer	line 8	Java Problem
BeanDefinitionOverrideFailureAnalyzer cannot be resolved to a type	_FactoryProvider.java	/webmvc-tomcat/target/spring-aot-test/src/main/java/org/springframework/boot/diagnostics/analyzer	line 9	Java Problem
BeanDefinitionOverrideFailureAnalyzer cannot be resolved to a type	_FactoryProvider.java	/webmvc-tomcat/target/spring-aot/src/main/java/org/springframework/boot/diagnostics/analyzer	line 8	Java Problem
```

The target/spring-aot and target/spring-aot-test folders are added as source folders but that looks wrong, it should be target/spring-aot/src/main/java  and target/spring-aot-test/src/main/java. If I manually edit them to be those values, I am down to 155 errors, like this:
```
Description	Resource	Path	Location	Type
AutoConfigureDataCassandra cannot be resolved to a type	StaticSpringFactories.java	/webmvc-tomcat/target/spring-aot-test/src/main/java/org/springframework/aot	line 166	Java Problem
AutoConfigureDataLdap cannot be resolved to a type	StaticSpringFactories.java	/webmvc-tomcat/target/spring-aot-test/src/main/java/org/springframework/aot	line 169	Java Problem
AutoConfigureDataMongo cannot be resolved to a type	StaticSpringFactories.java	/webmvc-tomcat/target/spring-aot-test/src/main/java/org/springframework/aot	line 152	Java Problem
```

**Comment by sdeleuze** [comment]
Fixed by 40015bb82fa2de2b29ce530fbd770088b78c8d0c and tested.
